addons/sale_purchase/tests/test_sale_purchase.py

```diff
@@ -258,3 +259,19 @@ def test_pol_description(self):
 
         po = self.env['purchase.order'].search([('partner_id', '=', self.partner_vendor_service.id)], order='id desc', limit=1)
         self.assertEqual(po.order_line.name, "[C01] Name01")
+
+    def test_cancel_sale_order(self):
+        """ Confirm SO, cancel it by Sale user (without permissions in the purchasing)"""
+        self.env.user.groups_id = [(6, 0, self.env.ref("sales_team.group_sale_salesman").ids)]
+        self.sale_order_1.action_confirm()
+
+        purchase_order = self.env['purchase.order'].sudo().search([('partner_id', '=', self.supplierinfo1.partner_id.id), ('state', '=', 'draft')])
+        purchase_lines = self.env['purchase.order.line'].sudo().search([('sale_line_id', 'in', self.sale_order_1.order_line.ids)])
+
+        self.assertEqual(len(purchase_lines), 1, "Only one purchase line should be created on SO confirmation")
+        self.assertEqual(len(purchase_order), 1, "One purchase order should have been created on SO confirmation")
+
+        wizard = Form(self.env['sale.order.cancel'].with_context({'default_order_id': self.sale_order_1.id})).save()
```

Comment:
Thanks 🙂 

We could also check `display_purchase_orders_alert` contains the expected value before calling `action_cancel` method. 🙂 

After that, I will let @odoo/rd-logistics doing the final review. 🙂

Reply:
@xavierbol okey, done